io-engine/src/rebuild/rebuild_descriptor.rs

```diff
@@ -85,9 +100,168 @@ impl RebuildDescriptor {
 
     /// Marks the rebuild segment starting from the given logical block as
     /// already transferred.
+    #[inline(always)]
     pub(super) fn blk_synced(&self, blk: u64) {
         if let Some(map) = self.rebuild_map.lock().as_mut() {
             map.blk_clean(blk);
         }
     }
+
+    /// Returns `IoVec` for the givem `DmaBuf`, with length adjusted to the copy
+    /// size for the given offset. Given `DmaBuf` must be large enough.
+    #[inline(always)]
+    pub(super) fn adjusted_iov(
+        &self,
+        buffer: &DmaBuf,
+        offset_blk: u64,
+    ) -> IoVec {
+        let mut iov = buffer.to_io_vec();
+
+        let iov_len = self.get_segment_size_blks(offset_blk) * self.block_size;
```

Comment:
Do we need to do this if we now specify the num_blocks ? on read/writev_blocks_async?

Reply:
It turned that yes we have. I tried not doing that at first, and I/O failed in a strange way. I didn't look further but it looks like SPDK's APIs require num_blocks to be equal to iovecs size.